### PR TITLE
bug fixed

### PR DESCRIPTION
**Pull request by Keyshal-211**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?

Please describe the test strategy you followed. If not tested, please explain why.

- [X] automated unit test
- [ ] automated integration test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
    
## Checklist before requesting a review

- [X] My code is easy to understand
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] My code fulfills all acceptance criteria
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation or
  the [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I made no breaking changes in the database schema or if so, I will perform a database migration

## Checklist for reviewer

- The code is easy to understand
- The code follows
  the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of
  this project
- The code is tested or if not, the reason is documented or discussed
- The added and existing tests reasonably cover the code change
- The code has no breaking changes in the database schema or if so, the assignee is aware of it and
  will perform a database migration

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/content_service/service/TagService.java [hunk 1]
```diff
@@ -23,8 +23,11 @@ public void synchronizeTags(ContentEntity content, List<String> tagNames) {
         List<TagEntity> existingTagsWithGivenNames = tagRepository.findByNameIn(tagNames);
         TagSynchronizationResult result = prepareSynchronization(tagNames, currentTags, existingTagsWithGivenNames);
         synchronizeWithDatabase(content, result);
+        List<TagEntity> unusedTags = tagRepository.findUnusedTags();
+        for (TagEntity unusedTag : unusedTags) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/content_service/service/TagService.java:27]: I think there is also a "deleteAll" method, but I'm not sure

**Review by PaulBredl — CHANGES_REQUESTED** [review]
Please implement a test that proves you're solution is working.
As you not filled out the pull request template, I don't know if and how you've tested this

File: src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationDeleteContentTest.java [hunk 2]
```diff
@@ -83,10 +87,20 @@ void testDeleteExistingContent(GraphQlTester graphQlTester) {
         System.out.println(contentRepository.findAll());
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationDeleteContentTest.java:87]:
```suggestion
```

Comment (Keyshal-211) [reply src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationDeleteContentTest.java]: done

File: src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationRemoveTagFromContentTest.java [hunk 3]
```diff
@@ -64,6 +71,10 @@ void testRemoveTagFromContent(GraphQlTester graphQlTester) {
         ContentEntity updatedContentEntity = contentRepository.findById(contentEntity.getId()).orElseThrow();
         assertThat(updatedContentEntity.getMetadata().getTags(), hasSize(1));
         assertThat(updatedContentEntity.getTagNames(), containsInAnyOrder("tag2"));
+        List<TagEntity> tags=tagRepository.findAll();
+        assertThat(tags, hasSize(2));
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationRemoveTagFromContentTest.java:75]: if you're implementation works, the tags should have size 1

**Review by PaulBredl — CHANGES_REQUESTED** [review]
Your code does not compile. Did you even run the tests??

File: src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationDeleteContentTest.java [hunk 4]
```diff
@@ -81,12 +87,15 @@ void testDeleteExistingContent(GraphQlTester graphQlTester) {
                 .path("deleteContent").entity(UUID.class).isEqualTo(contentEntity.getId());
 
         assertThat(contentRepository.findById(contentEntity.getId()).isEmpty(), is(true));
-        System.out.println(contentRepository.findAll());
         assertThat(contentRepository.count(), is(0L));
-        // Tags are not deleted (yet)
-        assertThat(tagRepository.count(), is(2L));
+        //Test that tag is deleted
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationDeleteContentTest.java:91]: You also need to update the test below, it fails

File: src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationRemoveTagFromContentTest.java [hunk 5]
```diff
@@ -7,26 +7,33 @@
...
 @TablesToDelete({"content_tags", "content", "tag"})
 class MutationRemoveTagFromContentTest {
 
     @Autowired
     private ContentRepository contentRepository;
-
+    @Autowired
+    private TagRepository tagRepository;
+    @Autowired
+    private TagService tagServiceMock; // Mock the TagService
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationRemoveTagFromContentTest.java:36]: where do you need the tag service? I dont see any usages

Comment (Keyshal-211) [reply src/test/java/de/unistuttgart/iste/gits/content_service/api/mutation/MutationRemoveTagFromContentTest.java:36]: > where do you need the tag service? I dont see any usages

its from my previous code, it has no use. I will remove this

**Comment by Keyshal-211** [comment]
> Your code does not compile. Did you even run the tests??

yes, I run the tests

**Review by PaulBredl — APPROVED** [review]

**Pull request merged**